### PR TITLE
Add more context to service errors

### LINKED ISSUE #347: Improve service errors - add more context
The errors of the validation functions for the service's data are not good enough. They are not explicit enough.

https://github.com/mesg-foundation/core/blob/bd84ad1d54eadfb8aa2bd79ac0c7c68f775f9c03/service/error.go#L14-L18
- [x] Rename `Key` to `EventKey` to be more specific

https://github.com/mesg-foundation/core/blob/bd84ad1d54eadfb8aa2bd79ac0c7c68f775f9c03/service/error.go#L38-L43
- [x] Rename `Key` to `TaskKey` to be more specific

https://github.com/mesg-foundation/core/blob/bd84ad1d54eadfb8aa2bd79ac0c7c68f775f9c03/service/error.go#L53-L61
- [x] Add `TaskKey` to `OutputNotFoundError` error to give more context

https://github.com/mesg-foundation/core/blob/bd84ad1d54eadfb8aa2bd79ac0c7c68f775f9c03/service/error.go#L63-L76
- [x] Rename `Key` to `OutputKey` to be more specific
- [x] Add `TaskKey` to `InvalidOutputDataError` to give more context
- [x] `e.Key` is the outputKey, not the task key. The `errorString` has to be fixed and also include the add `TaskKey`.

### PR DESCRIPTION
**Pull request by NicolasMahe**
Fix #347

### CODE REVIEW

File: execution/complete.go [hunk 1]
```diff
@@ -14,13 +14,15 @@ func (execution *Execution) Complete(output string, data map[string]interface{})
...
 	}
 	if !serviceOutput.IsValid(data) {
 		return &service.InvalidOutputDataError{
-			Output: serviceOutput,
-			Key:    output,
-			Data:   data,
+			Output:    serviceOutput,
+			TaskKey:   execution.Task,
+			OutputKey: output,
+			Data:      data,
```
Comment (antho1404) [inline execution/complete.go]: why not `OutputData` in this case ? it's not coherent with the `OutputKey` now

File: event/event.go [hunk 2]
```diff
@@ -26,9 +26,9 @@ func Create(serviceForEvent *service.Service, eventKey string, data map[string]i
 	}
 	if !serviceEvent.IsValid(data) {
 		return nil, &service.InvalidEventDataError{
-			Event: serviceEvent,
-			Key:   eventKey,
-			Data:  data,
+			Event:    serviceEvent,
+			EventKey: eventKey,
+			Data:     data,
```
Comment (antho1404) [inline event/event.go]: why not `EventData` ?

File: execution/create.go [hunk 3]
```diff
@@ -19,9 +19,9 @@ func Create(serviceForExecution *service.Service, task string, inputs map[string
 	}
 	if !serviceTask.IsValid(inputs) {
 		return nil, &service.InvalidTaskInputError{
-			Task:   serviceTask,
-			Key:    task,
-			Inputs: inputs,
+			Task:    serviceTask,
+			TaskKey: task,
+			Inputs:  inputs,
```
Comment (antho1404) [inline execution/create.go]: `InputData` ?

File: service/error.go [hunk 4]
```diff
@@ -12,13 +12,13 @@ func (e *EventNotFoundError) Error() string {
 
 // InvalidEventDataError is an error when the data of an event are not valid.
 type InvalidEventDataError struct {
-	Event *Event
-	Key   string
-	Data  map[string]interface{}
+	Event    *Event
+	EventKey string
+	Data     map[string]interface{}
```
Comment (antho1404) [inline service/error.go]: `EventData`

File: service/error.go [hunk 5]
```diff
@@ -37,13 +37,13 @@ func (e *TaskNotFoundError) Error() string {
 
 // InvalidTaskInputError is an error when the inputs of a task are not valid.
 type InvalidTaskInputError struct {
-	Task   *Task
-	Key    string
-	Inputs map[string]interface{}
+	Task    *Task
+	TaskKey string
+	Inputs  map[string]interface{}
```
Comment (antho1404) [inline service/error.go]: `InputData`

**Comment by NicolasMahe** [comment]
@ilgooz I wanted to do only a small PR to finish the 1.1 milestone. Let's continue on the v1.2 milestone.

**Pull request merged**